cdbot/cogs/maths.py

```diff
@@ -143,31 +142,55 @@ async def challenge(self, ctx: Context, number: int = 1):
         return await ctx.send(embed=embed)
 
     @command()
-    async def latex(self, ctx: Context, expression: str):
-        """Render a LaTeX expression."""
+    async def latex(self, ctx: Context, *, expression: str):
+        """
+        Render a LaTeX expression with https://quicklatex.com/
+        """
+        await self.latexRender(ctx, expression)
+
+    async def latexRender(self, ctx: Context, expression: str):
         channel = ctx.channel.id if type(ctx) is Context else ctx.id
 
         if channel in constants.BLOCKED_CHANNELS:
             return await ctx.send(
                 "\N{NO ENTRY SIGN} You cannot use this command in this channel!", delete_after=10
             )
 
-        options = {
-            "auth": {"user": "guest", "password": "guest"},
-            "latex": expression,
-            "resolution": 900,
-            "color": "969696",
-        }
+        # Code and regexes taken from https://quicklatex.com/js/quicklatex.js
+        # aiohttp seems to URL-encode things in a way quicklatex doesn't like
+
+        if expression.startswith("...latex") or expression.startswith(":latex"):
+            return
```

Comment:
When invoked with the command, `expression` won't contain the command prefix. You can remove these lines.

Reply:
the issue here is that, if those lines are removed, latex_render will fire twice on an expression like `:latex here's some maths $x^2$` -- once because of the `:latex` prefix (the function will be called with `here's some maths $x^2$`) and again because of the inline $ detector regex (the function will be called with `:latex here's some maths $x^2$`). I don't want to remove the :latex function, as it can be handy if the user wants to have something in LaTeX that doesn't include any $ (e.g. align blocks), so this felt like the easiest way to stop the function from firing twice

Reply:
So what's happening with this then?

Reply:
@Sh3llcod3 it feels to me as though the easiest solution is to keep it as-is